core/src/helpers.rs

```diff
@@ -169,16 +172,281 @@ pub fn starknet_block_to_eth_block(block: MaybePendingStarknetBlock) -> RichBloc
+                    timestamp,
+                    difficulty,
+                    nonce,
+                    size,
+                };
+                let block = Block {
+                    header,
+                    total_difficulty,
+                    uncles: vec![],
+                    transactions,
+                    base_fee_per_gas: None,
+                    size,
+                };
+                Rich::<Block> {
+                    inner: block,
+                    extra_info: BTreeMap::default(),
+                }
             }
-            MaybePendingBlockWithTxs::Block(_block_with_txs) => {
-                unimplemented!("BlockWithTxs is not supported yet")
+            MaybePendingBlockWithTxs::Block(block_with_txs) => {
+                let hash = H256::from_slice(&block_with_txs.block_hash.to_bytes_be());
+                let parent_hash = H256::from_slice(&block_with_txs.parent_hash.to_bytes_be());
+                let sequencer =
+                    H160::from_slice(&block_with_txs.sequencer_address.to_bytes_be()[12..32]);
+                let state_root = H256::from_slice(&block_with_txs.new_root.to_bytes_be());
+                let number = U256::from(block_with_txs.block_number);
+                let timestamp = U256::from(block_with_txs.timestamp);
+                let transactions = BlockTransactions::Full(
+                    block_with_txs
+                        .transactions
+                        .into_iter()
+                        .map(from_starknet_transaction_ethers_transaction)
+                        .collect(),
+                );
+                let header = Header {
+                    hash: Some(hash),
+                    parent_hash,
+                    uncles_hash: parent_hash,
+                    author: sequencer,
+                    miner: sequencer,
+                    state_root,
+                    // BlockWithTxHashes doesn't have a transactions root
+                    transactions_root: H256::zero(),
+                    // BlockWithTxHashes doesn't have a receipts root
+                    receipts_root: H256::zero(),
+                    number: Some(number),
+                    gas_used,
+                    gas_limit,
+                    extra_data,
+                    logs_bloom,
+                    timestamp,
+                    difficulty,
+                    nonce,
+                    size,
+                };
+                let block = Block {
+                    header,
+                    total_difficulty,
+                    uncles: vec![],
+                    transactions,
+                    base_fee_per_gas: None,
+                    size,
+                };
+                Rich::<Block> {
+                    inner: block,
+                    extra_info: BTreeMap::default(),
+                }
             }
         },
     }
 }
 
+fn from_starknet_transaction_ethers_transaction(tx: StarknetTransaction) -> EtherTransaction {
```

Comment:
how about `from_starknet_tx_to_eth_tx` 
or `starknet_tx_into_eth_tx` 
i.e. we don't need "ethers" with an s
we don't need to fully spell transaction imho

Reply:
Much better haha

Reply:
I like starknet_tx_into_eth_tx 
(Im bad at naming btw)